### PR TITLE
Fix inability of rpm-ostree to install multiple packages

### PR DESCRIPTION
**Pull request by happz**
That's 3 lines removed. The rest of 800+ lines of the patch consist of various improvements to related tests:

* new unit test, uses package managers directly to install multiple packages,
* `prepare/install` test refactored and extended to cover more distros when testing the basic operations, through parametrization,
* added new custom images to make tests cover "ostree" scenario
* plenty of `TODO` comments that should be resolved later.

Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

**Comment by lukaszachy** [comment]
So this PR now depends on https://github.com/teemtee/tmt/pull/2828, right?

File: containers/fedora/coreos/Containerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#
+# A Fedora CoreOS image tailored for tmt test suite
+#
+
+FROM quay.io/fedora/fedora-coreos:stable
+
+# Inject `dnf` to make things more complicated for `rpm-ostree` package
+# manager implementation
+RUN rpm-ostree install dnf5 \
+    && dnf5 remove -y diffutils
```
Comment (thrix) [inline containers/fedora/coreos/Containerfile:10]: @happz why is this needed?

File: containers/fedora/coreos/ostree/Containerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#
+# A Fedora CoreOS image tailored for tmt test suite
+#
+# Similar to `Containerfile.coreos`, but aims at ostree environment
```
Comment (thrix) [inline containers/fedora/coreos/ostree/Containerfile:4]:
```suggestion
# Similar to `Containerfile.coreos`, but aims to simulate booted environment
```

it is all ostree based really

File: containers/fedora/rawhide/Containerfile.unprivileged [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# A custom Fedora rawhide image, with unprivieged account & password-less sudo
+# A Fedora rawhide image tailored for tmt test suite, with unprivieged account & password-less sudo
```
Comment (thrix) [inline containers/fedora/rawhide/Containerfile.unprivileged:2]:
```suggestion
# A Fedora rawhide image tailored for tmt test suite, with unprivileged account & password-less sudo
```

File: tests/prepare/install/main.fmf [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+    Also verify special characters are correctly escaped, and copr
+    functionality.
+duration: 120m
 tag+:
   - provision-only
   - provision-container
-  - provision-local
   - provision-virtual
+
+# TODO: what test cases are safe enough to be executed against the loalhost?
```
Comment (thrix) [inline tests/prepare/install/main.fmf:13]:
```suggestion
# TODO: what test cases are safe enough to be executed against the localhost?
```

**Review by thrix — APPROVED** [review]
Just nitpicks

File: containers/fedora/coreos/Containerfile (hunk 1 above)
Comment (happz) [reply containers/fedora/coreos/Containerfile:13]: Added a comment in ea82cf1f

File: containers/fedora/coreos/ostree/Containerfile (hunk 2 above)
Comment (happz) [reply containers/fedora/coreos/ostree/Containerfile]: Fixed in ea82cf1f

File: containers/fedora/rawhide/Containerfile.unprivileged (hunk 3 above)
Comment (happz) [reply containers/fedora/rawhide/Containerfile.unprivileged]: Fixed in ea82cf1f

File: tests/prepare/install/main.fmf (hunk 4 above)
Comment (happz) [reply tests/prepare/install/main.fmf]: Fixed in ea82cf1f

File: tests/prepare/install/data/empty.fmf [hunk 5]
```diff
@@ -0,0 +1,8 @@
+summary: Do nothing in prepare, CLI will provide inputs
+prepare: []
+
+adjust:
+  - when: distro == ubuntu
+    prepare+:
+      - how: shell
+        script: apt-get update
```
Comment (psss) [inline tests/prepare/install/data/empty.fmf:8]: I'd say it really should not be user responsibility to call `apt-get update`. I'd suggest to add it to the `apt ` installer code instead. Even if it would be called every time before the `apt install` command (for now, to simplify the implementation). To me this approach just does not seem to be right. But, that would be definitely for a separate pull request.

**Review by psss — APPROVED** [review]
Looks good, thanks for substantially improving the test coverage! Just two (non-blocking) thoughts.

File: tests/prepare/install/data/empty.fmf (hunk 5 above)
Comment (happz) [reply tests/prepare/install/data/empty.fmf:8]: True, it will end up in tmt code, consider this as a remnant of times when the borderline was not as clear as it is now (i.e. any time before the Tuesday meeting this week :)

File: tests/prepare/install/main.fmf [hunk 6]
```diff
@@ -1,11 +1,14 @@
...
+description: |
+    Check basic installation of required and recommended packages, both
+    valid and invalid ones.
+    Also verify special characters are correctly escaped, and copr
+    functionality.
+duration: 120m
 tag+:
   - provision-only
   - provision-container
-  - provision-local
```
[the remark this answers is not in the document]
Comment (happz) [reply tests/prepare/install/main.fmf:10]: Yeah, but what does that mean? Would the test try to install something on localhost? Do we want that when the test tries to run against various environments through container and VM?

I can restore this branch, modified to fit the refactored tests, of course, I just want to be sure possible modification of localhost is expected and desired (and meaningful).

File: tests/prepare/install/data/empty.fmf (hunk 5 above)
Comment (psss) [reply tests/prepare/install/data/empty.fmf:8]: Ack.

File: tests/prepare/install/main.fmf (hunk 6 above)
[the remark this answers is not in the document]
Comment (psss) [reply tests/prepare/install/main.fmf:10]: Let's handle that in a separate PR in `1.34` as agree on chat.
